packages/govuk-frontend/src/govuk/common/index.mjs

```diff
@@ -133,6 +133,46 @@ export function getFragmentFromUrl(url) {
   return url.split('#').pop()
 }
 
+/**
+ * Move focus to element
+ *
+ * Sets tabindex to -1 to make the element programmatically focusable,
+ * but removes it on blur as the element doesn't need to be focused again.
+ *
+ * @private
+ * @template {HTMLElement} FocusElement
+ * @param {FocusElement} $element - HTML element
+ * @param {object} [options] - Handler options
+ * @param {function(this: FocusElement): void} [options.onBeforeFocus] - Callback before focus
+ * @param {function(this: FocusElement): void} [options.onBlur] - Callback on blur
+ */
+export function setFocus($element, options = {}) {
+  const isFocusable = $element.getAttribute('tabindex')
+
+  if (!isFocusable) {
+    $element.setAttribute('tabindex', '-1')
+  }
+
+  /**
+   * Reset element focus
+   */
+  function resetFocus() {
+    options.onBlur?.call($element)
+
+    if (!isFocusable) {
+      $element.removeAttribute('tabindex')
+    }
+  }
+
+  // Add listener to reset element on blur
+  $element.removeEventListener('blur', resetFocus)
```

Comment:
**issue** Reading your last comment on the listeners, I looked a bit more closely at this part and noticed something I missed. 

I think this may not do what we intend it to do actually. The `resetFocus` that gets passed to `removeEventListener` is not the same reference as the one that would have been added by a previous `addEventListener` inside `setFocus`, like in [this Codepen demo](https://codepen.io/romaricpascal/pen/yLZdojK?editors=1011).

Reply:
**thought** This makes me wonder if we actually do need all that cleanup:
* Keeping the `tabindex` in doesn't hurt anything, the element will remain programmatically focusable, but not focusable through other means (clicks, keyboard...) so won't impact users
* For the SkipLink, if we constantly remove the class on `blur`, I don't think that's a big deal that a listener remains on the linked element all the time. We can also have it registered during the `focus` listener with `{once: true}` so it's only running on the next blur ([support looks OK for browsers with ES Modules](https://browsersl.ist/#q=supports+es6-module+and+not+supports+once-event-listener)). Could do the same for the `focus` if we want to and only run it on the next `focus` after `setFocus` is called.

Reply:
@romaricpascal Cheers for the demo, yeah I understand references and it's totally my fault for last minute changes to keep file size down 🙈

I'm very glad you like `{ once: true }` though, let's do that instead

Otherwise it's quite noticeable without the cleanup when Error summary boxes are click-focusable

Reply:
I've revised commit 1a56bfd31ca59022c87bedeb665217f88217ca62 with `{ once: true }` so `removeEventListener()` has gone

In terms of "teardown" it would be nice to fully clean up after ourselves, but there's one other difference between the components we could perhaps discuss from https://github.com/alphagov/govuk-frontend/issues/4465?

1. Skip links preserve the hash fragment on click
2. Tab links preserve the hash fragment on click
3. Error summary links [prevent the hash fragment on click](https://github.com/alphagov/govuk-frontend/blob/ed08beadef553575a24eefcda41f0eed85c53b36/packages/govuk-frontend/src/govuk/components/error-summary/error-summary.mjs#L67-L72)

The issue highlights that hash fragments stick around after redirects:

* https://github.com/alphagov/govuk-frontend/issues/4465

Reply:
File size has increased from ~69.9 KiB~ 69.44 KiB → 69.5 KiB which is still a saving

Reply:
> In terms of "teardown" it would be nice to fully clean up after ourselves, but there's one other difference between the components we could perhaps discuss from #4465?

I think we can look at #4465 on its own, so we can focus just on the hash fragments.